src/io/WKTReader.cpp

```diff
@@ -210,60 +254,62 @@ std::unique_ptr<Geometry>
 WKTReader::readGeometryTaggedText(StringTokenizer* tokenizer) const
 {
     std::string type = getNextWord(tokenizer);
-    if(type == "POINT") {
-        return readPointText(tokenizer);
+
+    OrdinateSet ordinateFlags = readOrdinateFlags(type);
+
+    if(isTypeName(type, "POINT")) {
+        return readPointText(tokenizer, ordinateFlags);
     }
-    else if(type == "LINESTRING") {
-        return readLineStringText(tokenizer);
+    else if(isTypeName(type, "LINESTRING")) {
+        return readLineStringText(tokenizer, ordinateFlags);
     }
-    else if(type == "LINEARRING") {
-        return readLinearRingText(tokenizer);
+    else if(isTypeName(type, "LINEARRING")) {
+        return readLinearRingText(tokenizer, ordinateFlags);
     }
-    else if(type == "POLYGON") {
-        return readPolygonText(tokenizer);
+    else if(isTypeName(type, "POLYGON")) {
+        return readPolygonText(tokenizer, ordinateFlags);
     }
-    else if(type == "MULTIPOINT") {
-        return readMultiPointText(tokenizer);
+    else if(isTypeName(type,  "MULTIPOINT")) {
+        return readMultiPointText(tokenizer, ordinateFlags);
     }
-    else if(type == "MULTILINESTRING") {
-        return readMultiLineStringText(tokenizer);
+    else if(isTypeName(type, "MULTILINESTRING")) {
+        return readMultiLineStringText(tokenizer, ordinateFlags);
     }
-    else if(type == "MULTIPOLYGON") {
-        return readMultiPolygonText(tokenizer);
+    else if(isTypeName(type, "MULTIPOLYGON")) {
+        return readMultiPolygonText(tokenizer, ordinateFlags);
     }
-    else if(type == "GEOMETRYCOLLECTION") {
-        return readGeometryCollectionText(tokenizer);
+    else if(isTypeName(type, "GEOMETRYCOLLECTION")) {
+        return readGeometryCollectionText(tokenizer, ordinateFlags);
     }
     throw ParseException("Unknown type", type);
 }
 
 std::unique_ptr<Point>
-WKTReader::readPointText(StringTokenizer* tokenizer) const
+WKTReader::readPointText(StringTokenizer* tokenizer, OrdinateSet ordinateFlags) const
```

Comment:
Are `ordinateFlags` deliberately pass-by-value? For multi-part geometries, sometimes knowing the tagged dimensionality is necessary to handle conflicts in dimensionality deeper down (`GEOMETRYCOLLECTION Z(POINT Z(0 0 0), LINESTRING M(1 1 1, 2 2 2))`)

Reply:
This PR ensures that valid WKT with Z/M is read correctly but does not ensure that invalid WKT throws an error (the full scope of https://trac.osgeo.org/geos/ticket/1080). I can add that behavior to this PR if it's desirable.

From that ticket it sounds like nested geometries with different coordinate dimensions should _not_ be allowed. In that case, I guess we want to set the ordinate flags after reading the first coordinate of the first geometry, then "lock" them and throw a `ParseException` if subsequent coordinates do not comply? (which would require `ordinateFlags` to be pass-by-reference, as you suggest)

Reply:
This is what PostGIS parser does. For tagged dimensionality, the tag rules. For no-tag case, expect consistency with the first coordinate. I suppose throwing would be a new behaviour, in case anyone is feeding mixed dimensionality into GEOS right now. Those poor unfortunate souls.

Reply:
I've updated the PR to enforce dimensionality within a geometry. I think this wraps up https://trac.osgeo.org/geos/ticket/1080.